mlflow/utils/docstring_utils.py

```diff
@@ -182,6 +182,34 @@ def decorator(func):
 
 :ref:`This example<pip-requirements-example>` demonstrates how to specify pip requirements using
 ``pip_requirements`` and ``extra_pip_requirements``.
+""",
+        "signature": """
+an instance of the :py:class:`ModelSignature <mlflow.models.ModelSignature>`
+class that describes the model's inputs and outputs. If not specified but an
+``input_example`` is supplied, a signature will be automatically inferred
+based on the supplied input example and model. To disable automatic signature
+inference when providing an input example, set ``signature`` to ``False``.
+To manually infer a model signature, call
+:py:func:`infer_signature() <mlflow.models.infer_signature>` on datasets 
+with valid model inputs, such as a training dataset with the target column
+omitted, and valid model outputs, like model predictions made on the training
+dataset, for example:
+
+.. code-block:: python
+
+    from mlflow.models import infer_signature
+
+    train = df.drop_column("target_label")
+    predictions = ...  # compute model predictions
+    signature = infer_signature(train, predictions)
```

Comment:
@dbczumar I'm wondering if we should omit this example from the `signature` doc strings, for a few reasons:
1. If users want to learn how to use `infer_signature`, they can just click on the `infer_signature()` link.
2. This example is not very helpful anyways. It isn't specific to the model flavor. Users would still need to figure out on their own how to compute model predictions for the flavor in question.
3. Calling `infer_signature` is no longer the preferred approach for getting a signature. We want users to use input examples instead.

Reply:
Landing this PR and preserving the status quo for now. We can choose whether to omit this code example in a follow-up PR.